docs/howtos/serviceingress.md

```diff
@@ -0,0 +1,109 @@
+```
+
+### Step 2: Create a key pair to act as your Certificate Authority (CA)
+
+```
+openssl req -x509 -sha256 -newkey rsa:4096 -keyout ca.key -out ca.crt -days 1000 -nodes -subj '/CN=My Cert Authority'
+```
+
+### Step 3: Create key pairs for the server and for the client and sign them with the CA
+
+```
+# create and sign the server keys
+openssl req -new -newkey rsa:4096 -keyout server.key -out server.csr -nodes -subj '/CN={the host of your server}' -addext "subjectAltName=DNS:{the host of your server}"
+
+openssl x509 -req -sha256 -days 1000 -in server.csr -CA ca.crt -CAkey ca.key -set_serial 01 -out server.crt
+
+# create and sign the client keys
+openssl req -new -newkey rsa:4096 -keyout client.key -out client.csr -nodes -subj '/CN=Client'
+
+openssl x509 -req -sha256 -days 1000 -in client.csr -CA ca.crt -CAkey ca.key -set_serial 02 -out client.crt
+```
+
+### Step 4: Create a Kubernetes Secret
+
+```
+kubectl create secret generic -n thundernetes-system tls-certs --from-file=tls.crt=server.crt --from-file=tls.key=server.key --from-file=ca.crt=ca.crt
+```
+
+### Step 5: Deploy the Service and the Ingress
+
+We have bundled the definitions to deploy the GameServer API in the [deployment/secured/deploy_mtls.yaml](https://github.com/PlayFab/thundernetes/blob/main/cmd/gameserverapi/deployment/secured/deploy_mtls.yaml) file , it includes the Deployment, the Service, and the Ingress needed for the mTLS to work. You have to check the name of the Secret referenced in the Ingress matches the one you created, and also add the host values required. Then you just run:
+
+```
+kubectl apply -f {path to deploy_mtls.yaml}
+```
+
+### Step 6: Connect to the Game Server API
+
+Now the Game Server API is exposed through the Ingress, to connect to it you have to get the Ingress' external IP, you can do this with this command:
+
+```
+kubectl get ingress thundernetes-gameserverapi-ingress -n thundernetes-system
+```
+
+The Ingress may take a minute before getting an IP, if you're running this locally it won't ever get one, but you can use port forwarding instead. Finally you can try a simple GET request providing the client keys to test that the API is working, note that you will have to add the public certificate of your CA to your trusted root certificates:
+
+```
+curl https://{ingress_IP}/api/v1/gameserverbuilds --cert client.crt --key client.key
+```
+
+Or, you can also skip the validation on the client side and only check that the server is verifying the client certificates, adding the ```-k``` or ```--insecure``` option:
+
+```
+curl https://{ingress_IP}/api/v1/gameserverbuilds --cert client.crt --key client.key -k
+```
+
+### Extra: Using an IP instead of a domain
+
+Although it is not officially supported, you can use this configuration for an IP instead of a domain. The first thing to do is adding the IP of the ingress as a CN and SAN when you generate the server certificates:
```

Comment:
Is this going to be fairly public and copied? If so, we shouldn't likely add anything that isn't officially supported as we'll probably be on the hook for it later.

Reply:
I'd assume it refers to the nginx ingress supporting this configuration, not Thundernetes. @javier-op, any links on how/why this is not supported (e.g. an open issue?). It would be cool to include

Reply:
Yeah, I was kind of proud of figuring this out after a long googling session haha, but you're right they might blame it on us if something happens. I'm gonna remove it.